app/services/open_graph.rb

```diff
@@ -0,0 +1,77 @@
+# This class provides methods for getting structured metadata properties for
+# Open Graph usage
+#
+# @note A wrapper around the metainspector gem
+class OpenGraph
+  delegate :meta, :meta_tags, :images, to: :page
+
+  attr_accessor :page, :tags
+
+  DEFAULT_METHODS = %i(author title description).freeze
+
+  def initialize(url)
+    @page = MetaInspector.new(url)
+    @tags = meta_tags
+  end
+
+  # the optional "use_best" argument is used to determine if we should use the
+  # "best" option availabile for said method this works by picking the most
+  # available (first?) image, description, or author (respectively), if defined
+  # with the og:* or twitter:* metatags. example page.title(true) would look in
+  # said areas the default only looks at the head or default metatag.
+
+  DEFAULT_METHODS.each do |method_name|
+    define_method(method_name) do |use_best = false|
+      if use_best
+        page.public_send("best_#{method_name}".to_sym)
+      else
+        page.public_send(method_name)
+      end
+    end
+  end
```

Comment:
Looking for feedback here as I know meta-programmed methods can be frowned upon. What's the general thought here? If not this, we'd have three methods for `title`, `description`, and `author` that basically have the same logic. Do we favor perhaps more readability over being a little more DRY?

Reply:
I'm quite happy with meta programming.  We have 6 prior instances in app of calling `define_method`